### ISSUE TITLE
Export the 'Overlay settings' code to make it reusable

### ISSUE THREAD

**Issue by RahiDroid** [WordPress/gutenberg#44217]
## What problem does this address?
The `Post Featured Image` block's [Overlay element](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-featured-image/overlay.js), and the `Cover` block's Overlay element (divided between [InspectorControls](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/cover/edit/inspector-controls.js) and [Edit component](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/cover/edit/index.js)) have some similar code. Ideally, we'd want to avoid code repetition as much as possible. In future, too, if we happen to add this functionality to one more block, we shouldn't copy it over. This issue tries to address that. 

## What is your proposed solution?
I don't have a solution. I need opinions as to where such kind of code is to be put which could be shared across multiple blocks.

**Comment by RahiDroid** [comment]
@scruffian @mikachan 👋 

I have created a follow-up issue for the refactoring we discussed in https://github.com/WordPress/gutenberg/pull/43838#issuecomment-1242162826. I need thoughts on where such code should be exported so multiple blocks can share it. Please share your thoughts or tag relevant folks on this ticket. 

Thanks!

**Comment by RahiDroid** [comment]
This refactor should happen before we go about adding the overlay for the image block in https://github.com/WordPress/gutenberg/issues/44428

I will try and bring this ticket up in the next editor chat to get input on this.

**Comment by scruffian** [comment]
Could it go into "components"?
